### PR TITLE
Definitely specify the zero vector/quat's normalization behavior

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: #

### Changelog

* This fixes that for example `Vec3.normalize(v3(1, 2, 3), v3(0, 0, 0))` unexpectedly yields `v3(1, 2, 3)`.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/math/quat.ts [hunk 1]
```diff
@@ -408,6 +412,11 @@ export class Quat extends ValueType {
             out.y = a.y * len;
             out.z = a.z * len;
             out.w = a.w * len;
+        } else {
+            out.x = 0.0;
+            out.y = 0.0;
+            out.z = 0.0;
+            out.w = 1.0;
```
Comment (shrinktofit) [inline cocos/core/math/quat.ts:419]: Note here: differ fromvectors.

**Review by stanleyljl — APPROVED** [review]

**Comment by stanleyljl** [comment]
C++ implementation also needs to be modified together，and should be consistent with TS implementation.
I have no problem about ts implementation.

File: cocos/core/math/quat.ts [hunk 2]
```diff
@@ -397,8 +397,12 @@ export class Quat extends ValueType {
...
 
     /**
-     * @en Normalize the given quaternion
-     * @zh 归一化四元数
+     * @zh 归一化指定四元数。
+     * @en Normalizes the specified quaternion.
+     * @out @zh 结果四元数。@en Result quaternion.
+     * @param a @zh 输入四元数。@en Input quaternion.
+     * @note @zh 如果输入四元数 **精确** 等于零四元数，结果将设为单位四元数。
+     * @en If the input quaternion is **EXACTLY** zero vector, the quaternion will be set to identity quaternion.
```
Comment (minggo) [inline cocos/core/math/quat.ts:405]: I think it is better to describe what is `identity quaternion`. I am not sure if it is a well known words.

File: tests/value-types/quat.test.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { quat, Quat } from '../../cocos/core/math/quat';
```
Comment (minggo) [inline tests/value-types/quat.test.ts:1]: Why not add it in tests/core/math?

**Comment by shrinktofit** [comment]
Should replaced by https://github.com/cocos/cocos-engine/pull/14174

**Pull request closed without merging**